### PR TITLE
Docs: Replace DocCardLists with DocLinkLists

### PR DESCRIPTION
**Pull request by josmperez**
This PR utilizes the DocLinkLists added in https://github.com/grafana/plugin-tools/pull/493 to the site's index pages. Add two index pages for develop-a-plugin and extend-a-plugin.

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM

**Review by sympatheticmoose — CHANGES_REQUESTED** [review]
I don't think this works yet, doclinklist doesn't support folders.

This impacts:
- ~tutorials~
- create a plugin
- migration guides

We can discuss more with Jack when he's back

PR: Screenshot 2023-11-06 at 09 47 45

Live: 
Screenshot 2023-11-06 at 09 48 13

**Comment by jackw** [comment]
>how does it look for /migration-guides which has a mixture of folders and pages?

Right now it looks like the screenshot below which is a little misleading as it suggests the update create-plugin versions and migrate from toolkit are angular related. I can separate them in someway, just not entirely sure how off the top of my head. 😄 

image

**Comment by sympatheticmoose** [comment]
Figured that might be the case 😄 hmm, personally I would just go the titles and use of custom metadata for now. I'm not convinced its worth worrying too much over. 

> link to first item in category

To work around having to pick we can always just add the `_index` file with `DocLinkList` for the ones which lack it:
- upgrade from grafana versions
- develop a plugin
- extend a plugin

**Comment by jackw** [comment]
> hmm, personally I would just go the titles and use of custom metadata to indicate its a collection of content for now. I'm not convinced its worth worrying too much over.

Just to confirm, we'd be happy to add the customProps.description to each `_category_.json` and for it to look like this:

image

**Comment by sympatheticmoose** [comment]
I'm happy with that

**Comment by josmperez** [comment]
> we'd be happy to add the customProps.description to each `_category_.json` and for it to look like this:

looks good

**Comment by jackw** [comment]
@josmperez I've pushed a commit to this branch that updates the DocLinkList component as we've discussed above. To add descriptions in category links you'll need to update the `_category_.json` files with `customProps.description`. Example below:

```diff
{
  "position": 1,
  "label": "Develop a plugin",
+  "customProps": {
+    "description": "Category description text."
+  },
  "collapsible": true,
  "collapsed": true
}
```

File: docusaurus/docs/create-a-plugin/_category_.json
Comment (sympatheticmoose) [inline docusaurus/docs/create-a-plugin/_category_.json:1]: I don't think this is needed but theres no harm in having it.

@jackw I assume these are optional and only needed if the page/category is going to be included within a DocLinkList (which to my knowledge this isn't). Is there any other benefit to the metadata?

File: docusaurus/docs/create-a-plugin/extend-a-plugin/_category_.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {
-    "position": 2,
-    "label": "Extend a plugin",
-    "collapsible": true,
-    "collapsed": true
-  }
+  "position": 2,
+  "label": "Extend a plugin",
+  "customProps": {
+    "description": "Guides for extending a Grafana plugin with features such as authentication, query editor help, and annotations."
```
Comment (sympatheticmoose) [inline docusaurus/docs/create-a-plugin/extend-a-plugin/_category_.json:5]:
```suggestion
    "description": "Guides for extending your Grafana plugin with additional features to improve its quality."
```

File: docusaurus/docs/migration-guides/angular-react/_category_.json [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
-    "position": 10,
-    "label": "Migrate from AngularJS to React",
-    "collapsible": true,
-    "collapsed": true
-  }
\ No newline at end of file
+  "position": 10,
+  "label": "Migrate from AngularJS to React",
+  "customProps": {
+    "description": "How to migrate your plugin from Angular to ReactJS."
```
Comment (sympatheticmoose) [inline docusaurus/docs/migration-guides/angular-react/_category_.json:5]:
```suggestion
    "description": "How to migrate your plugin from the legacy AngularJS framework to our React based platform."
```

File: docusaurus/docs/migration-guides/update-from-grafana-versions/_category_.json [hunk 3]
```diff
@@ -1,6 +1,9 @@
...
-    "position": 1,
-    "label": "Update from Grafana versions",
-    "collapsible": true,
-    "collapsed": true
-  }
\ No newline at end of file
+  "position": 1,
+  "label": "Update from Grafana versions",
+  "customProps": {
+    "description": "How to update your plugin from one version of Grafana to another."
```
Comment (sympatheticmoose) [inline docusaurus/docs/migration-guides/update-from-grafana-versions/_category_.json:5]:
```suggestion
    "description": "How to handle breaking changes between Grafana versions in your plugin."
```

File: docusaurus/docs/tutorials/_category_.json
Comment (sympatheticmoose) [inline docusaurus/docs/tutorials/_category_.json:1]: Again, not sure we need this but no harm in keeping

**Review by sympatheticmoose — APPROVED** [review]
minor suggestions, no blockers 🚀

**Pull request merged**